api/include/opentelemetry/trace/propagation/HttpTraceContext.cc

```diff
@@ -0,0 +1,129 @@
+#include <vector>
+
+namespace trace
+{
+namespace propagation
+{
+namespace
+{
+class HttpTraceContext : public HTTPTextFormat
+{
+
+    private:
+        static const nostd::string_view TRACE_PARENT = "traceparent";
+        static const nostd::string_view TRACE_STATE = "tracestate";
+        static const std::vector<nostd::string_view> FIELDS = {TRACE_PARENT, TRACE_STATE)};
+        static const int VERSION_BYTES = 2;
+        static const int TRACE_ID_BYTES = 32;
+        static const int PARENT_ID_BYTES = 16;
+        static const int TRACE_FLAG_BYTES = 2;
+        static const int TRACE_DELIMITER_BYTES = 3;
+        static const int HEADER_SIZE = VERSION_BYTES + TRACE_ID_BYTES + PARENT_ID_BYTES + TRACE_FLAG_BYTES + TRACE_DELIMITER_BYTES;
+
+        static T checkNotNull(T arg, nostd::string_view errorMessage) {
+            if (arg == NULL) {
+                throw new NullPointerException(errorMessage);
+            }
+            return arg;
+        }
+
+        static void injectImpl(Setter setter, T &carrier, const SpanContext spanContext) {
+            char hex_string[HEADER_SIZE];
+            sprintf(hex_string, "00-%032x-%016x-%02x",span_context.trace_id,span_context.span_id,span_context.trace_flags);
+            nostd::string_view traceparent_string(hex_string); // I don't know if the string view has a function that takes in an char array convert it to string though.
```

Comment:
There is a nostd::string_view(const char *str).  Note that there's also a string_view(const char *str, size_t len) if the input isn't null terminated.